### DISCUSSION TITLE
Http::withBody() throw an error when send a file

### DISCUSSION THREAD

**Discussion by fransiskusbenny** [laravel/framework#33480] [General]
- Laravel Version: 7.19.0
- PHP Version: 7.4.7
- Database Driver & Version: -/-

### Description:
Http client throw an error when send  a file using ```withBody``` method

### Steps To Reproduce:

```php
$http = Http::withHeaders(['Content-Encoding' => 'gzip'])
            ->withBody(fopen('file.gz', 'r'), 'application/gzip')
            ->post('http://example.com');
```

```
Argument 1 passed to Illuminate\Http\Client\Request::withData() must be of the type array, resource given
```

**Comment by driesvints** [comment]
This sounds more like a feature request so I'm moving this to discussions.
